### ISSUE TITLE
Sed: Pages Not Saving

### ISSUE THREAD

**Issue by egbunse** [StraboSpot/strabo-mobile#484]
From Sed Field Trip: certain pages weren't saving data (they'd be filled out completely but then were blank upon returning to the page again)- 'Sed Lithologies- Basics' and 'Sed Interpretations- Environmental Interpretation.' This happened on my Android phone and the iPad Mini.

**Comment by jessicagood** [comment]
@egbunse This sounds pretty bad but I can't figure out how to recreate this. Every time I try, including on Android and iPad, it always saves the data. Any other help you can give me or the exact steps to reproducing this error?

**Comment by egbunse** [comment]
@jessicagood I just realized on the 'Sed Lithologies-Basics' page that I was trying to add an interval to the Spot holding an entire Strat Section, so no wonder it wasn't saving (i.e.- I set a Spot on the MapView and toggled on 'Add Stratigraphic Section' then went over to Sed Lithologies-Basics instead of adding the interval with the '+' on 'View Stratigraphic Section'- just wasn't thinking). 
To remedy this, I noticed that the 'Strat Section' page isn't available within a Strat Section Interval Spot. It might make sense for the Whole Strat Section Spot to not have the 'Sed Lithologies,' 'Sed Structures,' and 'Sed Interpretations' pages available since these seem to be things we fill out for Interval Spots.
This would also help the 'Sed Interpretations-Environmental Interpretation' save problem I had too. I was trying to fill that page out for a normal Spot that was not an interval in a Strat Section. 
@UASTeam what do you think? Am I confused on the workflow here?

**Comment by jessicagood** [comment]
@egbunse I actually had it that way initially, where you could only see the 'Sed Lithologies,' 'Sed Structures,' and 'Sed Interpretations' pages if you were in an Interval, but I was asked to change it so the user could see those pages if they're toggled on no matter what type of Spot they're in. It sounded like they wanted the option for someone to still record this information for a Spot even if they weren't creating a strat section. Though I agree with you this could be confusing, I'm not sure if there is some better way to organize this. The way we have it now does gives more flexibility for different workflows.

**Comment by egbunse** [comment]
@jessicagood I've been trying to add info to these pages in normal spots and the interface either freezes up (clicking to go to other pages or the map doesn't work) and/or when revisiting the info entered isn't saved. 
Also, maybe to avoid confusion on the 'Sed Lithologies' page, the basics page should have a note or message box to only use this page in association with an interval spot?

**Comment by jessicagood** [comment]
@egbunse Ok, that helped me find a few related bugs for the freezing or not saving, hopefully fixed with 2b9a41828f6dcc0d297af3f1e0b11608dffcf468. I updated the testing app. Can you check it out?

The problem with a note on the Basics page that this page is only to be used in association with an interval spot is that they want to be able to use the page with ANY Spot. If a user doesn't want to create a strat section but still wants to put in sed data then that's why the page exists for any Spot if the toggle for that page is turned on. At least that is how I understood it. We can talk about this in the meeting tomorrow if you think it should be designed differently.

**Comment by egbunse** [comment]
@jessicagood Great! I just was able to successfully add the Sed Interpretations- Environment data that I had taken a screenshot of in the field. Was able to go back to the Spots list and see that the 'Images' and 'Sed Interpretations' pages were filled out for that Spot. Then I made a new Spot and filled out 'Sed Lithologies,' 'Sed Structures,' and 'Sed Interpretations' and they all saved, so I'm going to close this issue.
Also, now that you explain that, it makes sense. What you describe essentially what I wanted to do- I made a spot for the top of a roadcut (we were doing surficial geology at that point) that we extensively discussed (so I needed those specific sed pages), but I didn't make a strat column/section because we didn't spend much time contemplating the bottom exposure.
